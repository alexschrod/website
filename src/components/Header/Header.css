header {
	position: relative;
	width: 100%;
	min-height: 100%;
	text-align: center;
	color: #fff;
	background-image: url(header.jpg);
	background-position: center;
	-webkit-background-size: cover;
	-moz-background-size: cover;
	background-size: cover;
	-o-background-size: cover;
}

header.small {
	background: #222;
	position: static;
	width: 100%;
	min-height: 50px;
	max-height: 50px;
}

header.small * {
	display: none;
}

header .header-content {
	position: relative;
	width: 100%;
	padding: 100px 15px;
	text-align: center;
}

header .header-content .header-content-inner h1 {
	margin-top: 30px;
	margin-bottom: 0;
	font-weight: 700;
	font-size: 65px;
}

header .header-content .header-content-inner hr {
	margin: 30px auto;
}

header .header-content .header-content-inner p {
	margin-bottom: 50px;
	font-size: 16px;
	font-weight: 300;
	color: rgba(255, 255, 255, .7);
}

@media(min-width:768px) {
	header .header-content {
		position: absolute;
		top: 50%;
		padding: 0 50px;
		-webkit-transform: translateY(-50%);
		-ms-transform: translateY(-50%);
		transform: translateY(-50%);
	}
	header .header-content .header-content-inner {
		margin-right: auto;
		margin-left: auto;
		max-width: 1000px;
	}
	header .header-content .header-content-inner p {
		margin-right: auto;
		margin-left: auto;
		max-width: 80%;
		font-size: 18px;
	}
}
