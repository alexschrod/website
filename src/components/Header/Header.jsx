import React from "react"
import containerStyles from "./Header.module.css"
import portrait from "./portrait.jpg"

const Header = ({ small, header, tagline }) => {
  if (small) {
    return <header className={containerStyles.headerSmall}></header>
  } else {
    return (
      <header className={containerStyles.header}>
        <div className={containerStyles.headerContent}>
          <div className={containerStyles.headerContentInner}>
            <img className={containerStyles.imgCircle} src={portrait} alt="" />
            <h1 className={containerStyles.headerContentInnerH1}>{header}</h1>
            <hr className={containerStyles.headerContentInnerHr} />
            <span className={containerStyles.btnBlue}>{tagline}</span>
          </div>
        </div>
      </header>
    )
  }
}

export default Header
