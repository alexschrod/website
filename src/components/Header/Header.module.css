.header {
  position: relative;
  width: 100%;
  min-height: 100%;
  text-align: center;
  color: #fff;
  background-image: url(./header.jpg);
  background-position: center;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  background-size: cover;
  -o-background-size: cover;
}

.headerSmall {
  composes: header;
  background: #222;
  position: static;
  width: 100%;
  min-height: 50px;
  max-height: 50px;
}

.headerContent {
  position: relative;
  width: 100%;
  padding: 100px 15px;
  text-align: center;
}

.headerContentInnerH1 {
  margin-top: 30px;
  margin-bottom: 0;
  font-weight: 700;
  font-size: 65px;
}

.headerContentInnerHr {
  composes: blue from global;
  margin: 30px auto;
}

.headerContentInner p {
  margin-bottom: 50px;
  font-size: 16px;
  font-weight: 300;
  color: rgba(255, 255, 255, 0.7);
}

.imgCircle {
  display: block;
  margin: auto;
  border-radius: 70%;
  max-width: 280px;
}

.btnBlue {
  padding: 15px 30px;
  font-size: 2em;
  border: 0;
  border-radius: 300px;
  text-transform: uppercase;
  font-weight: 700;
  color: #fff;
  background-color: #4075f0;
  display: inline-block;
  line-height: 1.42857143;
}

@media (min-width: 768px) {
  .headerContent {
    position: absolute;
    top: 50%;
    padding: 0 50px;
    -webkit-transform: translateY(-50%);
    -ms-transform: translateY(-50%);
    transform: translateY(-50%);
  }
  .headerContentInner {
    margin-right: auto;
    margin-left: auto;
    max-width: 1000px;
  }
  .imgCircle {
    max-width: 200px;
  }
}
