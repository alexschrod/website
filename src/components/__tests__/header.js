import React from "react"
import renderer from "react-test-renderer"

import Header from "../Header"

const header = "Header"
const tagLine = "Tag Line"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Header header={header} tagline={tagLine} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("small renders correctly", () => {
    const tree = renderer.create(<Header small />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
