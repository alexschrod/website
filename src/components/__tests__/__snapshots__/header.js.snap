// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="header"
>
  <div
    className="headerContent"
  >
    <div
      className="headerContentInner"
    >
      <img
        alt=""
        className="imgCircle"
        src="test-file-stub"
      />
      <h1
        className="headerContentInnerH1"
      >
        Header
      </h1>
      <hr
        className="headerContentInnerHr"
      />
      <span
        className="btnBlue"
      >
        Tag Line
      </span>
    </div>
  </div>
</header>
`;

exports[`Header small renders correctly 1`] = `
<header
  className="headerSmall"
/>
`;
